<!-- This should eventually launch either simulated or hardware arms -->

<launch>
    
    <arg name="machine" default=""/>
    <arg name="display_machine" default=""/>
    <arg name="ns" default="left"/>
    <arg name="launch_rviz" default="false"/>
    <arg name="x_pos" default="0"/>

    <group ns="$(arg ns)">

        <rosparam param="IK_timeout">0.07</rosparam>

        <include file="$(find ar3_gazebo)/launch/ar3L_gazebo_bringup.launch">
            <arg name="launch_world" value="false"/>
            <arg name="ns" value="$(arg ns)"/>
            <arg name="x_pos" value="$(arg x_pos)"/>
            <!--<arg name="machine" value="$(arg machine)"/>-->
        </include>



        <node name="world_transform" pkg="ar3_gazebo" type="tf_world_transform.py"/>

        <include file="$(find ar3_moveit_config)/launch/ar3_moveit_bringup.launch">
            <arg name="launch_rviz" value="$(arg launch_rviz)"/>
            <arg name="machine" value="$(arg machine)"/>
            <arg name="display_machine" value="$(arg display_machine)"/>
            <arg name="ns" value="$(arg ns)"/>
            <!--<arg name="simulated" value="true"/>-->
        </include>

        <node name="move_interface" pkg="marsha_core" type="move_interface" respawn="false" output="screen" machine="$(arg machine)"/>
        

        <node name="get_state" pkg="marsha_core" type="get_state" machine="$(arg machine)" if="$(eval arg('launch_rviz') == true)"/>
        
        <include file="$(find marsha_core)/launch/gripper.launch">
            <arg name="machine" value="$(arg machine)"/>
            <arg name="ns" value="$(arg ns)"/>
        </include>
    </group>


</launch>